### PR TITLE
add usrsctp/0.9.4.0 recipe

### PR DESCRIPTION
**Pull request by triangulumlabs**
Specify library name and version:  **usrsctp/master**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Make sure to follow the bots instructions to gain access to the EAP

**Comment by triangulumlabs** [comment]
> Make sure to follow the bots instructions to gain access to the EAP

Yes,
I've already submitted EAP approval.

**Comment by uilianries** [comment]
please, don't forget to sign the CLA too: https://github.com/conan-io/conan-center-index/pull/4134#issuecomment-755914315

File: recipes/usrsctp/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+sources:
+  "master":
```
Comment (uilianries) [inline recipes/usrsctp/all/conandata.yml:2]: Use official releases only. If it's too old, ask to the upstream for a new one. When abandoned, you can create a version based on the commit date: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

File: recipes/usrsctp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+from shutil import copyfile
+import os
+
+
+class ConanRecipe(ConanFile):
+    name = "usrsctp"
+    description = "SCTP user-land implementation (usrsctp)"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/sctplab/usrsctp"
+    license = "BSD 3"
```
Comment (uilianries) [inline recipes/usrsctp/all/conanfile.py:11]:
```suggestion
    license = "BSD-3-Clause"
```

https://spdx.org/licenses/BSD-3-Clause.html

File: recipes/usrsctp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["sctp_build_programs"] = self.options.programs
+        cmake.definitions["sctp_build_fuzzer"] = self.options.fuzzer
+        cmake.configure(source_folder=self._source_subfolder)
```
Comment (uilianries) [inline recipes/usrsctp/all/conanfile.py:50]:
```suggestion
        cmake.configure()
```

If you point to source_subfolder, the cmake wrapper will be ignored.

File: recipes/usrsctp/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        "fPIC": [True, False],
+        "programs": [True, False],
+        "fuzzer": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "programs": False,
+        "fuzzer": False,
+    }
```
Comment (uilianries) [inline recipes/usrsctp/all/conanfile.py:24]:
```suggestion
    }
    exports_sources = "CMakeLists.txt"
```
You have to export your cmake wrapper

File: recipes/usrsctp/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cxx
```
Comment (uilianries) [inline recipes/usrsctp/all/conanfile.py:38]:
```suggestion
        del self.settings.compiler.libcxx
```

File: recipes/usrsctp/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["sctp_build_programs"] = self.options.programs
+        cmake.definitions["sctp_build_fuzzer"] = self.options.fuzzer
+        cmake.configure(source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```
Comment (uilianries) [inline recipes/usrsctp/all/conanfile.py:55]:
```suggestion
```

No need as there are no patches

File: recipes/usrsctp/all/test_package/test_package.c [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	while (usrsctp_finish() != 0)
+	{
+#ifdef _WIN32
+		Sleep(1000);
+#else
+		sleep(1);
+#endif
+	}
+	return (0);
+}
```
Comment (uilianries) [inline recipes/usrsctp/all/test_package/test_package.c:60]:
```suggestion
}

```

Add empty line

Comment (triangulumlabs) [reply recipes/usrsctp/all/test_package/test_package.c:60]: Thank you for very helpful suggestions @uilianries

**Comment by prince-chrismc** [comment]
> please, don't forget to sign the CLA too: [#4134 (comment)](https://github.com/conan-io/conan-center-index/pull/4134#issuecomment-755914315)

This is still missing in the status

**Pull request closed without merging**